src/Service/DescartarCarrinhoService.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Service;
+
+use App\Controller\Exception\CarrinhoJaDescartadoException;
+use App\Entity\Carrinho;
+use App\Entity\StatusEnum;
+use App\Repository\CarrinhoRepository;
+
+class DescartarCarrinhoService
+{
+  public function __construct(
+    private CarrinhoRepository $carrinhoRepository,
+  ) {}
+
+  public function execute(int|Carrinho $carrinho): void
+  {
+    $carrinho = $this->carrinhoRepository->find($carrinho);
+
+    if ($carrinho->getStatus() === StatusEnum::descartado) {
+      throw new CarrinhoJaDescartadoException('carrinho ja foi descartado', 500);
+    }
+
+    $dataAtual = new \DateTime();
+    if ($carrinho->getCriadoEm()->format('d/m/Y') !== $dataAtual->format('d/m/Y') && $carrinho->getStatus() === StatusEnum::aberto) {
+      $carrinho->setStatus(StatusEnum::descartado);
+      $carrinho->updateAtualizadoEm();
+      $carrinho->updateFinalizadoEm();
```

Comment:
cria um método assim dentro do Carrinho e joga esse código dentro dele...

```suggestion
      $carrinho->descartar();
```